src/types/siteInfo.ts

```diff
@@ -8,3 +10,20 @@ export type SiteInfo = {
   stagingUrl: StagingPermalink
   siteUrl: ProdPermalink
 }
+
+export interface DNSRecord {
+  source: string
+  type: keyof typeof RedirectionTypes
+  target: string
+}
+
+export interface DnsResultsForSite {
+  dnsRecords: DNSRecord[]
+  siteUrl: string
+}
+
+export interface SiteLaunchDto {
+  siteStatus: "LAUNCHED" | "NOT_LAUNCHED" | "LAUNCHING"
+  dnsRecords?: DNSRecord[] // only present iff siteStatus is LAUNCHED
+  siteUrl?: string
+}
```

Comment:
can we make different types for this? 

```ts
export type SiteLaunchDto = LaunchedSiteDto 
| SiteLaunchingDto
| SiteNotLaunchedDto
```

allows the inference to work better so that only if we assert `siteDto.siteStatus === "LAUNCHED"` then `dnsRecords` is present.

Reply:
14c038ff941fdc02c43c696236b65257c57b665c
oops this comment is wrong, as it could be in the `LAUNCHED` status but the records are still being generated